state-chain/amm/src/limit_orders.rs

```diff
@@ -787,10 +787,22 @@ impl<LiquidityProvider: Clone + Ord> PoolState<LiquidityProvider> {
 			let positions = &mut self.positions[!SD::INPUT_SIDE];
 			let fixed_pools = &mut self.fixed_pools[!SD::INPUT_SIDE];
 
-			let position = positions
-				.get(&(sqrt_price, lp.clone()))
-				.ok_or(PositionError::NonExistent)?
-				.clone();
+			let position = match positions.get(&(sqrt_price, lp.clone())) {
+				Some(position) => position.clone(),
+				None =>
+					if amount == u128::MAX.into() {
+						// We are removing the order anyway so not being able to
+						// find it is not a problem (this happens if implicit sweeping
+						// pre-removes the order if it was fully executed).
+						return Ok((
+							Default::default(),
+							Default::default(),
+							PositionInfo::default(),
+						));
```

Comment:
Isn't the expectation that this should return some non-zero collected amount? 

ie. whatever was swept during the auto-sweep...

Reply:
No, whatever was collected during `inner_sweep` is already in the strategy's balance at this point (however this gets rolled back due to the error later which this change was addressing).